pkg/operator/resourcesynccontroller/resourcesync_controller_test.go

```diff
@@ -249,6 +249,86 @@ func TestSyncConfigMap(t *testing.T) {
 	}
 }
 
+func TestSyncConditionally(t *testing.T) {
+	for _, tc := range []struct {
+		name       string
+		fn         preconditionsFulfilled
+		expectSync bool
+	}{
+		{
+			name:       "should sync when condition is fulfilled",
+			fn:         conditionFulfilled,
+			expectSync: true,
+		},
+		{
+			name:       "should not sync when condition is not fulfilled",
+			fn:         conditionNotFulfilled,
+			expectSync: false,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			kubeClient := fake.NewSimpleClientset(
+				&corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{Namespace: "config", Name: "secret"},
+				},
+				&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{Namespace: "config", Name: "configmap"},
+				},
+			)
+
+			configInformers := informers.NewSharedInformerFactoryWithOptions(kubeClient, 1*time.Minute, informers.WithNamespace("config"))
+			operatorInformers := informers.NewSharedInformerFactoryWithOptions(kubeClient, 1*time.Minute, informers.WithNamespace("operator"))
+
+			fakeStaticPodOperatorClient := v1helpers.NewFakeOperatorClient(
+				&operatorv1.OperatorSpec{
+					ManagementState: operatorv1.Managed,
+				},
+				&operatorv1.OperatorStatus{},
+				nil,
+			)
+			eventRecorder := events.NewRecorder(kubeClient.CoreV1().Events("test"), "test-operator", &corev1.ObjectReference{})
+
+			c := NewResourceSyncController(
+				fakeStaticPodOperatorClient,
+				v1helpers.NewFakeKubeInformersForNamespaces(map[string]informers.SharedInformerFactory{
```

Comment:
nit: I think you could create `ResourceSyncController` directly and avoid passing/creating informers

Reply:
we can't because we need to watch the source and destination namespaces to synchronise resources: https://github.com/openshift/library-go/blob/master/pkg/operator/resourcesynccontroller/resourcesync_controller.go#L131-L136